lib/pengine/bundle.c

```diff
@@ -1434,6 +1435,20 @@ print_rsc_in_list(resource_t *rsc, const char *pre_text, long options,
     }
 }
 
+static void
+output_rsc_in_list(resource_t *rsc, long options, pcmk__output_t * out)
+{
+    if (rsc != NULL) {
+        if (options & pe_print_html) {
+            out->begin_list(out, "li", NULL, NULL);
+        }
+        rsc->fns->output(rsc, options, out);
+        if (options & pe_print_html) {
+            out->end_list(out);
+        }
```

Comment:
The nice thing about the output model is that we don't need different behavior for different formats. I think you can get rid of this function altogether and just call rsc->fns->output() directly.

There's no way in the output model to optionally wrap something as a list item. Instead, you'd have to pass an option to rsc->fns->output() indicating whether it should be a list item, and define the output functions accordingly. Basically the output model is semantic rather than functional, e.g. you say something like "output a fencing event" rather than "output this prefix, then output this string, then end the line".

Reply:
I got the point, that there is a generic `pcmk__output_t` construct that takes the data and smart enough to print it in any required format. Although `output_rsc_in_list` function is superfluous, I think there could be some difficulties trying to teach the construct to understand the semantics of the data to print it accordingly. I understand we use the messages to hint the construct, but I cannot figure out, how `pcmk__output_t` will know that those data are list items if we don't say it explicitly.

Reply:
> you'd have to pass an option to rsc->fns->output() 
Should I add another parameter or encapsulate encapsulate it in existing, e.g. add a bit-flag in options, or pass it together with the rsc/out arguments?

Reply:
What if I use a message method eg. `out->message(out, "list-item",...)` that hints the system, it was a list item?